alphagov/paas-product-page
# [#151015901] Product page update & onboarding

# What

This PR is the result of several stories that have gotten a bit mixed up, but the main changes are

* Merges in many of the product page tweaks that have been floating around in the `paas-proto` experiments for some time.
* Migrates the backend from a static site to a dynamic ruby/sinatra app to enable us to do form submission and API calls to things like Deskpro. (This one done by using the previously created paas-submit application as the backend and letting it serve the erb files). 
* Adds the new "Getting Started" onboarding flow
* Adds a form/wizard for users to request a PaaS account
* Adds a generic support/contact-us form for creating tickets

# Why is this not squashed!?

Because the path to squashery is not yet clear.

...and I am expecting changes from review before I make it prettier.

# How to review

* Ensure you can run the app locally: `make dev`
* Ensure you can run the tests `make test` and they pass
* You may find it easier to code review this application in it's entirety at this point rather than per commit. So..
    * Code review of the sinatra backend (app.rb)
    * Check for any of the common web app mistakes (escaping/injection stuff in templating etc)
    * 
* Proof read content, especially on:
    * `/` (home)
    * `/signup`
    * `/contact-us`
* Ensure forms work (you will need to run the app locally with deskpro keys - see readme)
    * ...with js enabled: forms should display as 'wizard' with validation as you go
    * ...with js disabled: forms should display as a single big form with validation on submit
* There is a PR for updating the product-page deployment in release-ci: https://github.com/alphagov/paas-release-ci/pull/38

# Who can review

not @chrisfarms nor @samcrang nor @stephenjoe1
